### ISSUE TITLE
Linter warns on unused for-of variable

### ISSUE THREAD

**Issue by shicks** [google/closure-compiler#2341]
The following is a pain to do in any other way when `iter` is an arbitrary iterable:

```js
let count = 0;
for (const unused of iter) count++;
```

But the linter warns that `unused` is never read.  Possible workarounds would be (1) don't warn about unused variables whose names start with "unused", (2) possibly restrict this check to for-of variables.

**Comment by Yannic** [comment]
I think a nice was of handling this would be to introduce a single variable _ for unused values which cannot be read. 
```javascript
let count = 0;
for (_ of iter) count++;
```

**Comment by Dominator008** [comment]
"_" shadows the popular library underscore.js so I would not use that. "unused" seems like a good heuristic to me.

**Comment by Yannic** [comment]
I thought about `_` because it is used in Go for exactly this and using something similar sounded like a good idea to me, but you're right, it's not an option since it is used by a popular library and would break a lot of code. I'd still prefer a single name over a prefix
